### PR TITLE
Rebase to WildFly 11

### PR DESCRIPTION
**Pull request by bobmcwhirter**
Includes:

* Elytron subsystem.
* Elytron security provider for client tests.
* module-rewrite.conf/module.conf changes to accomodate the
  Weld module splits.
* Dealing with some deprecations from jboss-modules.
* Adjusting to the removal of selfcontained/ from wildly-core.
* Add some activators to ensure the infinispan caches start.
* Update to Keycloak.

- [X] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [X] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [X] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

**Comment by scottmarlow** [comment]
I'm guessing that the MongoDB failure in https://ci.wildfly-swarm.io/job/wildfly-swarm-docker-test-linux/270/display/redirect, means that I need some NoSQL subsystem changes.  I've had wildfly-nosql working with WF11 before, so shouldn't be too bad.

**Comment by bobmcwhirter** [comment]
If you want to PR against the wf11 branch we can wait to merge.

**Comment by scottmarlow** [comment]
I fetched wf11, I'll investigate the nosql test failure cause as a start.

**Comment by scottmarlow** [comment]
Do I need to rebuild wildfly-config-api locally with WildFly 11 or something?  I got a build error with your wf11 branch:

> Failed to execute goal on project spi: Could not resolve dependencies for project org.wildfly.swarm:spi:jar:2017.9.0-SNAPSHOT: Failure to find org.wildfly.swarm:config-api-runtime:jar:1.2.0-SNAPSHOT in https://repository.jboss.org/nexus/content/groups/public/ was cached in the local repository, resolution will not be reattempted until the update interval of jboss-public-repository-group has elapsed or updates are forced

**Comment by scottmarlow** [comment]
thanks, I was fetching from https://github.com/bobmcwhirter/wildfly-swarm wf11, oops :)

File: core/container/src/main/java/org/wildfly/swarm/Swarm.java [hunk 1]
```diff
@@ -740,6 +748,7 @@ public static void main(String... args) throws Exception {
             vme.printStackTrace();
             System.exit(1);
         } catch (final Throwable t) {
+            t.printStackTrace();
             if (System.getProperty("swarm.inhibit.auto-stop") == null) {
                 t.printStackTrace();
```
Comment (Ladicek) [inline core/container/src/main/java/org/wildfly/swarm/Swarm.java]: If we do it unconditionally now, two lines above, maybe wanna remove this one?

File: pom.xml [hunk 2]
```diff
@@ -1287,9 +1294,11 @@
           <name>!swarm.docs.skip</name>
         </property>
       </activation>
+<!--
       <modules>
         <module>docs</module>
       </modules>
+-->
```
Comment (Ladicek) [inline pom.xml]: Hmm not really :-)

File: core/container/src/main/java/org/wildfly/swarm/Swarm.java (hunk 1 above)
Comment (bobmcwhirter) [reply core/container/src/main/java/org/wildfly/swarm/Swarm.java]: No, it was my debug that I forgot to remove.

File: pom.xml (hunk 2 above)
Comment (bobmcwhirter) [reply pom.xml]: Good catch.

**Comment by scottmarlow** [comment]
I should have a PR later today for the Mongo test failure.

**Comment by scottmarlow** [comment]
I fixed the problem and released wildfly-nosql 1.0.0.Alpha6, please merge https://github.com/wildfly-swarm/wildfly-swarm/pull/717 to resolve the MongoDB test failure.  Thanks! :)

**Comment by bobmcwhirter** [comment]
DO NOT SQUASH as it contains commits from multiple committers.

**Comment by bobmcwhirter** [comment]
retest this please

**Comment by bobmcwhirter** [comment]
retest this please

**Comment by kenfinnigan** [comment]
retest this please

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**